backend/wmg/data/snapshot.py

```diff
@@ -122,20 +123,59 @@ def load_snapshot(
     will simply return the cached snapshot if there isn't a newer snapshot id.
     """
     global cached_snapshot
-
-    should_reload, snapshot_id = _should_reload_snapshot(
-        snapshot_schema_version=snapshot_schema_version,
-        explicit_snapshot_id_to_load=explicit_snapshot_id_to_load,
-        read_versioned_snapshot=read_versioned_snapshot,
-    )
-
-    if should_reload:
-        cached_snapshot = _load_snapshot(
+    if os.getenv("DEPLOYMENT_STAGE") is None:
```

Comment:
If DEPLOYMENT_STAGE is not set, assume we are running the script locally. When run locally, the load_snapshot function reads the latest snapshot ID from WMG_BUCKET_PATH, downloads the snapshot, and loads it into the WmgSnapshot interface. When run locally, WMG_BUCKET_PATH needs to be set (e.g. s3://cellxgene-wmg-prod). If neither DEPLOYMENT_STAGE nor WMG_BUCKET_PATH are set, then an error will be raised.

Reply:
this is mostly for aesthetic reasons but can you move all this into `_load_local_snapshot` so the hacky stuff is separate? logically it also seems like everything here should belong in `_load_local_snapshot`

Reply:
I kept it out of _load_local_snapshot because that function is meant to be able to load any local snapshot, not just the latest one from WMG_BUCKET_PATH